GliaWidgets/View/Chat/Message/Content/Text/ChatTextContentView.swift

```diff
@@ -101,11 +101,30 @@ extension ChatTextContentView: UITextViewDelegate {
         in characterRange: NSRange,
         interaction: UITextItemInteraction
     ) -> Bool {
-        if URL.scheme == "tel" || URL.scheme == "mailto" {
-            return true
+        if URL.scheme == "tel" {
+            openPhoneDialer(url: URL)
+        } else if URL.scheme == "mailto" {
+            openEmailClient(url: URL)
         } else {
             linkTapped?(URL)
-            return false
+        }
+
+        return false
+    }
+
+    private func openPhoneDialer(url: URL) {
+        let phone = url.absoluteString.replacingOccurrences(of: "tel:", with: "")
+
+        if let callUrl = URL(string: "tel://\(phone)"), UIApplication.shared.canOpenURL(callUrl) {
+            UIApplication.shared.open(callUrl)
```

Comment:
Let's introduce Environment with `UIApplication` struct injected. It can be introduced under `UIKitBased` namespace.
Also let's make `openPhoneDialer` and `openEmailClient` internal and add unit tests for them using introduced `Environment` and `UIApplication` structs.

Reply:
reworked as suggested by yuri. Does it make sense to still write test for openUrl?

Reply:
Yes, to make sure that openUrl will receive only `tel:` and `mailto:` schemas.

Reply:
added tests in cad98bcb8c746c71e72b9cdae0523f4518a16f21